shoop/themes/classic_gray/forms.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import re
+from collections import defaultdict
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from shoop.xtheme.plugins.forms import GenericPluginForm
+
+
+class SocialMediaLinksPluginForm(GenericPluginForm):
+    """
+    Form for the social media links xtheme plugin. One field is provided for each
+    entry in the plugin's icon_classes attribute, which maps social media site names
+    to font-awesome icon classes by default.
+    """
+    def populate(self):
+        """
+        Populates form with default plugin fields as well as any social media link type
+        included in the plugin's ``icon_classes`` attribute.
+
+        Also adds an ordering field for each link type to change display order.
+        """
+        icon_classes = self.plugin.icon_classes
+        links = self.plugin.config.get("links", {})
+        for name, icon_class in sorted(icon_classes.items()):
+            url = links[name]["url"] if name in links else ""
+            ordering = links[name]["ordering"] if name in links else None
+            self.fields[name] = forms.URLField(
+                label=_(name), required=False,
```

Comment:
How is `_(name)` supposed to work? It doesn't. `_("message")` has two functions: (1) It translates the message from the source string to currently active language when called and (2) it marks the message for translation. Because of 2, the argument must always be a string literal, never a variable.

=> Move the underscore calls as part of the dict definition.